src/staging/src/pages/alerts.js

```diff
@@ -1,133 +1,385 @@
+    } else if (advisoryType === "flood") {
+      q += "?eventType.eventType_contains=flood";
+    }
+
+    const apiCall = apiUrl + q;
 
-  if (typeof window !== "undefined" && window.document) {
-    thisUrl = new URLSearchParams(window.location.search);
-    params = Object.fromEntries(thisUrl.entries());
+    axios
+      .get(apiCall)
+      .then(function (data) {
+        setAdvisoryCount(data.data);
+      })
+
+      
   }
 
-  if (params && params.type) {
-    switch (params.type) {
-      case 'wildfires':
-        advisories = getWildfires(data.allStrapiPublicAdvisory.nodes);
-        pageTitle = 'Public Advisories | Wildfires';
-        advisoryType = 'wildfire ';
-        break;
-      case 'floods':
-        advisories = getFloods(data.allStrapiPublicAdvisory.nodes);
-        pageTitle = 'Public Advisories | Flooding';
-        advisoryType = 'flood ';
-        break;
-      default:
-        advisories = data.allStrapiPublicAdvisory.nodes;
+  const getApiQuery = useCallback((advisoryTypeFilter) => {
+
+    let q = "?_sort=advisoryDate:DESC"; // Order by date
+
+    if (advisoryTypeFilter === "wildfire") {
+      q += "&eventType.eventType_contains=wildfire";
+    } else if (advisoryTypeFilter === "flood") {
+      q += "&eventType.eventType_contains=flood";
+    }
+
+    let useParksFilter = isParksFilter;
+    let useTypesFilter = isTypesFilter;
+    let useKeywordFilter = isKeywordFilter;
+
+    // check if any checkbox filter is set
+    let anyFilter = isParksFilter || isTypesFilter || isKeywordFilter;
+     
+    if (!anyFilter) {
+      // use all filters if none are selected
+      useParksFilter = true;
+      useTypesFilter = true;
+      useKeywordFilter = true;
+    }
+
+    if (advisoryTypeFilter !== 'public') {
+      // filter is already applied
+      // ignore types checkbox
+      useTypesFilter = false;
+    }
+
+    // check if there is anything in the search textbox
+    let anySearch = (searchText && (searchText !== ""));
+
+    if (anySearch) { // only apply filter if there is a keyword
+
+      let n = 0;
+
+      if (useParksFilter) {
+        q += "&_where[_or][" + n + "]"
+        q += "[protectedAreas.protectedAreaName_contains]=" + searchText;
+        n++;
+      }
+      if (useTypesFilter) {
+        q += "&_where[_or][" + n + "]"
+        q += "[eventType.eventType_contains]=" + searchText;
+        n++;
+      }
+      if (useKeywordFilter) {
+        q += "&_where[_or][" + n + "]"
+        q += "[description_contains]=" + searchText;
+        n++;
+        q += "&_where[_or][" + n + "]"
+        q += "[title_contains]=" + searchText;
+      }
+    }
+
+    return (q);
+  }, [isKeywordFilter, isParksFilter, isTypesFilter, searchText]);
+
+  const getAdvisories = useCallback((q) => { // q = api query
+
+    let newApiCall = apiUrl + `/public-advisories` + q;
+
+    newApiCall += "&_limit=" + pageLen; // use -1 for unlimited
+    newApiCall += "&_start=" + (pageLen * (pageIndex - 1));
```

Comment:
These can be passed to axios as GET params instead of adding them manually, e.g. `.get(newApiCall,  { params: { _limit: pageLen, ...etc } })`

Reply:
for sure. if we remember, let's do this for next related ticket